portfolio/src/main/webapp/script.js

```diff
@@ -12,14 +12,128 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var map;
+var locationsDict = new Object();
+
+/** Calls all necessary functions to be called onLoad */
+function initialize() {
+  loadComments();
+  createMap();
+}
+
+/** Creates a map with all components and adds it to the page. */
+function createMap() {
+  map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: 41.8996099, lng: -87.6232902, h: -7, p: 22 },
+    zoom: 13,
+  });
+
+  setupMarkers();
+  changeView(locationsDict["Bean"]);
+}
+
+/** All setup code for markers. */
+function setupMarkers() {
+  locationsDict["Bean"] = {
+    m: null,
+    lat: 41.8826099,
+    lng: -87.6232902,
+    h: -7,
+    p: 22,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Cloud Gate</p>',
+    iw: null,
+  };
+  locationsDict["Skyline"] = {
+    m: null,
+    lat: 41.9147471,
+    lng: -87.6209924,
+    h: 185,
+    p: 17,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Photography Point</p>',
+    iw: null,
+  };
+  locationsDict["Riverwalk"] = {
+    m: null,
+    lat: 41.8877563,
+    lng: -87.6273952,
+    h: 125,
+    p: 30,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Chicago Riverwalk</p>',
+    iw: null,
+  };
+
+  for (var location in locationsDict) {
+    locationsDict[location].m = createMarker(locationsDict[location]);
+    locationsDict[location].iw = new google.maps.InfoWindow({
+      content: locationsDict[location].is,
+    });
+    addMarkerListeners(
+      locationsDict[location].m,
+      location,
+      locationsDict[location].iw
+    );
+  }
+}
+
+/** Add bounce animation and view listeners. */
+function addMarkerListeners(marker, place, infowindow) {
+  google.maps.event.addListener(marker, "click", function () {
+    changeView(locationsDict[place]);
+  });
+  google.maps.event.addListener(marker, "mouseover", function () {
+    marker.setAnimation(google.maps.Animation.BOUNCE);
+  });
+  google.maps.event.addListener(marker, "mouseout", function () {
+    marker.setAnimation(null);
+  });
+  marker.addListener("mouseover", function () {
+    infowindow.open(map, marker);
+  });
+  marker.addListener("mouseout", function () {
+    infowindow.close(map, marker);
+  });
+}
+
+/** Change street view to position in positionTuple. */
+function changeView(positionTuple) {
+  var panorama = new google.maps.StreetViewPanorama(
+    document.getElementById("pano"),
+    {
+      position: { lat: positionTuple.lat, lng: positionTuple.lng },
+      pov: {
+        heading: positionTuple.h,
+        pitch: positionTuple.p,
+      },
+    }
+  );
+  map.setStreetView(panorama);
+}
+
+/** Initialize and set up individual markers. */
+function createMarker(positionTuple) {
+  console.log("Created marker");
+  var tmp = new google.maps.Marker({
```

Comment:
use const or let (see https://google.github.io/styleguide/jsguide.html#features-use-const-and-let)